lib/utils/utf8.c

```diff
@@ -79,3 +84,34 @@ char *utf8_lcpy(char *dst, const char *src, size_t n)
 
 	return dst;
 }
+
+bool utf8_is_valid(const char *str, size_t maxlen)
+{
+	size_t i = 0, nbyte = 0;
+	size_t len = strnlen(str, maxlen);
+
+	while (i < len) {
+		if (str[i] <= ASCII_CHAR && str[i] >= '\0') {
+			i++;
+			continue;
+		} else {
+			if (str[i] <= SEQUENCE_MAX_LEN_2_BYTE &&
+				str[i] >= SEQUENCE_MIN_LEN_2_BYTE) {
+				nbyte = 2;
+			} else if (str[i] <= SEQUENCE_MAX_LEN_3_BYTE &&
+					   str[i] >= SEQUENCE_MIN_LEN_3_BYTE) {
+				nbyte = 3;
+			} else if (str[i] <= SEQUENCE_MAX_LEN_4_BYTE &&
+					   str[i] >= SEQUENCE_MIN_LEN_4_BYTE) {
+				nbyte = 4;
+			} else {
+				return false;
+			}
+		}
```

Comment:
This else is redundant, please move code to the left:
```suggestion
		}
		if (str[i] <= SEQUENCE_MAX_LEN_2_BYTE &&
			str[i] >= SEQUENCE_MIN_LEN_2_BYTE) {
			nbyte = 2;
		} else if (str[i] <= SEQUENCE_MAX_LEN_3_BYTE &&
				   str[i] >= SEQUENCE_MIN_LEN_3_BYTE) {
			nbyte = 3;
		} else if (str[i] <= SEQUENCE_MAX_LEN_4_BYTE &&
				   str[i] >= SEQUENCE_MIN_LEN_4_BYTE) {
			nbyte = 4;
		} else {
			return false;
		}
```

Reply:
It was over 100 columns and I had to wrap. So, I need to resolve this comment.

Reply:
Last time: Do not resolve other peoples comments!

I do not follow your logic. Moving code to the left makes the 100 column "issue" less, not worse. Please explain.

Reply:
I changed it to 
```
str[i] <= SEQUENCE_MAX_LEN_4_BYTE && str[i] >= SEQUENCE_MIN_LEN_4_BYTE) 
```
and it made CI throw out that the code exceeded 100 lines.

Reply:
> If you want me to give you an explanation, I will change it and you can look at the warnings CI throws.

Please post a link to where the CI is failing and I'll have a look.

> And it's just a look, it's not important. We should focus on the functionality of the decoder, not other.

One thing does not rule out the other. We can have both.

Reply:
Okay, I will revise it today. Please look forward to it ;)